pkg/k8s/init.go

```diff
@@ -271,3 +280,101 @@ func RegisterCRDs() error {
 
 	return nil
 }
+
+// WaitForCRD waits for the given CRD to be available until the default timeout,
+// after which cilium-agent should be ready. Returns an error when timeout
+// is exceeded.
+func WaitForCRD(client clientset.Interface, name string) error {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), option.Config.CRDWaitTimeout)
+	defer cancelFunc()
+	return waitForCRD(ctx, client, name)
+}
+
+// waitForCRD waits for the given CRD to be available with the given context.
+func waitForCRD(ctx context.Context, client clientset.Interface, name string) error {
+	log.Infof("Waiting for CRD %s to be available", name)
+
+	selector := fields.OneTermEqualSelector("metadata.name", name).String()
+	w := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			options.FieldSelector = selector
+
+			if k8sversion.Capabilities().APIExtensionsV1CRD {
+				return client.ApiextensionsV1().CustomResourceDefinitions().List(ctx, options)
+			}
+
+			return client.ApiextensionsV1beta1().CustomResourceDefinitions().List(ctx, options)
```

Comment:
we should make the `client.ApiextensionsV1beta1` conditional based on the k8s version that supports the v1 vs v1beta1.

Reply:
Not sure if I'm following, but we are doing this a few lines above.

Reply:
:man_facepalming: for some reason I read `ApiextensionsV1beta1` on both of them.